ghc-typelits-knownnat build failure

```
[1 of 3] Compiling GHC.TypeLits.KnownNat.Solver ( src/GHC/TypeLits/KnownNat/Solver.hs, dist/build/GHC/TypeLits/KnownNat/Solver.o )

src/GHC/TypeLits/KnownNat/Solver.hs:261:52: error:
    • Couldn't match type ‘(TcType.TcTyVar, TcType.TcType)’
                     with ‘Var.Var’
      Expected type: [(TcType.TcTyVar, TcType.TcType)]
        Actual type: [((TcType.TcTyVar, TcType.TcType), Ct)]
    • In the first argument of ‘substType’, namely ‘subst’
      In the expression: substType subst z
      In the expression: (x, y, substType subst z)
    |
261 |       kn_wanteds = map (\(x,y,z) -> (x,y,substType subst z))
    |                                                    ^^^^^
```

I was able to reproduce this locally like so:

```bash
stack unpack ghc-typelits-knownnat-0.4.2 && cd ghc-typelits-knownnat-0.4.2
edit stack.yaml # add the following stack.yaml
stack build --haddock --test --bench --no-run-benchmarks
```

```yaml
# stack.yaml
resolver: nightly-2018-05-08
extra-deps:
- ghc-tcplugins-extra-0.3
```

n.b. ghc-typelits-knownnat is being held back per #3597.

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -1673,8 +1673,8 @@ packages:
         - ghc-typelits-knownnat
         - ghc-typelits-natnormalise
         - clash-prelude
-        - clash-lib < 0 # via aeson-1.3.1.0
-        - clash-ghc < 0 # via clash-lib
+        - clash-lib
+        - clash-ghc
 
     "Athan Clark <athan.clark@gmail.com> @athanclark":
         - aeson-attoparsec
@@ -3431,9 +3431,6 @@ packages:
         - hmatrix-gsl < 0.19
         - hmatrix-special < 0.19
 
-        # https://github.com/fpco/stackage/issues/3544
-        - ghc-typelits-natnormalise < 0.6
-
         # https://github.com/fpco/stackage/issues/3559
         - parser-combinators < 1.0
 
@@ -3451,11 +3448,6 @@ packages:
 
         # https://github.com/fpco/stackage/issues/3596
         - brick < 0.37
-
-        # https://github.com/fpco/stackage/issues/3597
-        - ghc-typelits-knownnat < 0.5
-        - ghc-tcplugins-extra < 0.3 # https://github.com/fpco/stackage/issues/3599
-        - ghc-typelits-extra < 0.2.5
 
         # https://github.com/fpco/stackage/issues/3608
         - wl-pprint-text < 1.2

